posts/tree-cell-typing/index.qmd

```diff
@@ -0,0 +1,241 @@
+in the `InSituType` package. 
+
+```{r setup, message=FALSE, eval = FALSE}
+# Load libraries
+library(InSituType)
+library(InSituTree)
+
+# Read in io profiles
+data(ioprofiles)
+
+head(ioprofiles)
+```
+
+```{r, eval = TRUE, echo = FALSE}
+profile_head <- readRDS("./assets/io_head.rds")
+head(profile_head)
+```
+
+We start by setting up a nested list of cell types,
+visualized with the helper function `printTree()`.
+The nested list should have names for major cell types (for example,
+"B-lymphoid" or "structural" below), and should include all cell types in the
+reference profile as endpoints (for example, "B-cell" or "endothelial" below).
+The endpoint cell types do not need to be set up as lists (for example, see "NK"
+below which is not set as "NK" = "NK"), though setting them as lists will not
+cause errors.
+
+```{r setupCellTypeTree, message=FALSE, eval = FALSE}
+# Establish a nested list of cell types
+celltype_hierarchy_list <- list(
+  "structural" = c(
+    "endothelial", 
+    "fibroblast"),
+  "myeloid" = c(
+    "macrophage", 
+    "mast", 
+    "mDC", 
+    "monocyte", 
+    "neutrophil"),
+  "lymphoid" = list(
+    "B-lymphoid" = c(
+      "B-cell", 
+      "pDC",
+      "plasmablast"),
+    "T-lymphoid" = list(
+      "T4" = c(
+        "T CD4 memory", 
+        "T CD4 naive", 
+        "Treg"),
+      "T8" = c(
+        "T CD8 memory", 
+        "T CD8 naive"),
+      "NK")
+    )
+)
+
+printTree(celltype_hierarchy_list, prefix = "|-")
+```
+
+```{r, eval = TRUE, echo = FALSE, warning=FALSE}
+tree_text <- readLines("./assets/cth_tree.txt")
+cat(tree_text, sep = "\n")
+```
+
+
+Importantly, any middle nodes that have only a single child will
+act as end points for cell type assignments. See the example below:
+```{r poorCellTypeTree, eval = FALSE}
+# Example of a tree with a single child node
+cth_single_child_node <- list(
+  "structural" = c(
+    "endothelial", 
+    "fibroblast"),
+  "myeloid" = c(
+    "macrophage", 
+    "monocyte"),
+  "lymphoid" = list( # This is a single child node: all cells annotated as 
+    # lymphoid would also be 'B and T cells' and so annotations will not
+    # continue beyond this point.
+    "B and T cells" = list(
+      "B-lymphoid" = c(
+        "B-cell", 
+        "pDC",
+        "plasmablast"),
+      "T-lymphoid" = list(
+        "T4" = c(
+          "T CD4 memory", 
+          "T CD4 naive", 
+          "Treg"),
+        "T8" = c(
+          "T CD8 memory", 
+          "T CD8 naive"),
+        "NK")
+    )
+  )
+)
+
+printTree(cth_single_child_node, prefix = "|-")
+```
+
+```{r, eval = TRUE, echo = FALSE, warning=FALSE}
+tree_text <- readLines("./assets/cth_single_child_tree.txt")
+cat(tree_text, sep = "\n")
+```
+
+:::{.callout-tip}
+If you're working from a CosMx-derived cell profile from our [CosMx Cell Profile library](https://github.com/Nanostring-Biostats/CosMx-Cell-Profiles/tree/main){target="_blank"}, formatted nested cell type lists are available as `*celltypeslist.R` files.
```

Comment:
It might be better to have these 2 tips in same callout box.

Reply:
Merged to one tip.